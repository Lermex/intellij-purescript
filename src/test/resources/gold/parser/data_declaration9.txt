Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSDataDeclarationImpl(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('User9')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('b')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('User9')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(ObjectType)
            PsiElement({)('{')
            PsiElement(whitespace)('\n  ')
            PSASTWrapperElement(Row)
              PSASTWrapperElement(GenericIdentifier)
                PSIdentifierImpl(identifier)
                  PsiElement(identifier)('name')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(Type)
                PSASTWrapperElement(TypeAtom)
                  PSASTWrapperElement(GenericIdentifier)
                    PsiElement(identifier)('a')
                    PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(GenericIdentifier)
                      PsiElement(identifier)('b')
                      PsiElement(whitespace)('\n ')
            PsiElement(})('}')