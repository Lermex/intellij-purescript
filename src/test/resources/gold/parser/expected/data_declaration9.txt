Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSDataDeclarationImpl(DATA_DECLARATION)
        PsiElement(PSTokenType.data)('data')
        PSDeclarationTypeImpl(DECLARATION_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('User9')
        PSIdListImpl(ID_LIST)
          PSDeclarationIdImpl(DECLARATION_ID)
            PsiElement(PSTokenType.ID)('a')
          PSIdListImpl(ID_LIST)
            PSDeclarationIdImpl(DECLARATION_ID)
              PsiElement(PSTokenType.ID)('b')
        PsiElement(PSTokenType.=)('=')
        PSDataDeclarationItemListImpl(DATA_DECLARATION_ITEM_LIST)
          PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
            PSTypeListImpl(TYPE_LIST)
              PSDeclarationTypeImpl(DECLARATION_TYPE)
                PsiElement(PSTokenType.PROPER_NAME)('User9')
            PSIdListImpl(ID_LIST)
              PSDeclarationIdImpl(DECLARATION_ID)
                PsiElement(PSTokenType.ID)('a')
            PSDataDeclarationBodyImpl(DATA_DECLARATION_BODY)
              PsiElement(PSTokenType.{)('{')
              PSDataDeclarationTypeDefinitionListImpl(DATA_DECLARATION_TYPE_DEFINITION_LIST)
                PSDataDeclarationTypeDefinitionImpl(DATA_DECLARATION_TYPE_DEFINITION)
                  PsiElement(PSTokenType.ID)('name')
                  PsiElement(PSTokenType.::)('::')
                  PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                    PSDeclarationIdImpl(DECLARATION_ID)
                      PsiElement(PSTokenType.ID)('a')
                  PsiElement(PSTokenType.->)('->')
                  PSParenthesisedIdListImpl(PARENTHESISED_ID_LIST)
                    PSDeclarationIdImpl(DECLARATION_ID)
                      PsiElement(PSTokenType.ID)('b')
              PsiElement(PSTokenType.})('}')